folium/colormap.py

```diff
@@ -0,0 +1,594 @@
+
+    def render(self, **kwargs):
+        self.color_domain = [self.min + (self.max-self.min)*i/499. for i in range(500)]
+        self.color_range = [self.__call__(x) for x in self.color_domain]
+        self.tick_labels = legend_scaler(self.index)
+        self.caption = ""
+
+        super(ColorMap, self).render(**kwargs)
+
+        figure = self.get_root()
+        assert isinstance(figure, Figure), ("You cannot render this Element "
+                                            "if it's not in a Figure.")
+
+        figure.header.add_children(
+            JavascriptLink("https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.min.js"),  # noqa
+            name='d3')
+
+    def _rgba_floats_tuple(self, x):
+        """This class has to be implemented for each class inheriting from Colormap.
+        This has to be a function of the form float -> (float,float,float,float)
+        Describing for each input float x, the output color in RGBA format ;
+        each output value being between 0 and 1.
+        """
+        raise NotImplementedError
+
+    def rgba_floats_tuple(self, x):
+        """Provides the color corresponding to value `x` in the
+        form of a tuple (R,G,B,A) with float values between 0. and 1.
+        """
+        return self._rgba_floats_tuple(x)
+
+    def rgba_bytes_tuple(self, x):
+        """Provides the color corresponding to value `x` in the
+        form of a tuple (R,G,B,A) with int values between 0 and 255.
+        """
+        return tuple(int(u*255.9999) for u in self.rgba_floats_tuple(x))
+
+    def rgb_bytes_tuple(self, x):
+        """Provides the color corresponding to value `x` in the
+        form of a tuple (R,G,B) with int values between 0 and 255.
+        """
+        return self.rgba_bytes_tuple(x)[:3]
+
+    def rgb_hex_str(self, x):
+        """Provides the color corresponding to value `x` in the
+        form of a string of hewadecimal values "#RRGGBB".
+        """
+        return "#%02x%02x%02x" % self.rgb_bytes_tuple(x)
+
+    def __call__(self, x):
+        """Provides the color corresponding to value `x` in the
+        form of a string of hewadecimal values "#RRGGBB".
+        """
+        return self.rgb_hex_str(x)
+
+    def _repr_html_(self):
+        return (
+            '<svg height="50" width="500">'
+            + ''.join(
+                [(
+                    '<line x1="{i}" y1="0" x2="{i}" '
+                    'y2="20" style="stroke:{color};stroke-width:3;" />'
+                    ).format(
+                        i=i*1,
+                        color = self.rgb_hex_str(self.min + (self.max-self.min)*i/499.))
+                    for i in range(500)
+                    ])
+            + '<text x="0" y="35">{}</text>'.format(self.min)
+            + '<text x="500" y="35" style="text-anchor:end;">{}</text>'.format(self.max)
+            + '</svg>')
+
+class LinearColormap(ColorMap):
+    def __init__(self, colors, index=None, min=0., max=1., caption=""):
```

Comment:
Again let's consider changing all `min`/`max` to `vmin`/`vmax` to avoid name clashing.